spec/support/routes.rb

```diff
@@ -26,7 +26,9 @@ def draw_routes
     resources :returns
 
     scope "(/optional/:optional_id)" do
-      resources :things
+      resources :things do
+        get "deep/:deep_id(/:optional_deep_id)", to: "things#index", as: :deep
```

Comment:
I don't like parameters name here. The tests are harder to read because of it. Let's make a new route for your advanced tests like:

``` ruby
get (/:first_optional)/pages/:second_required/:third_required/edit(/:fourth_optional)
```

Better proposals welcome. I just want to make parameters order more logical so that you don't need to consult with route to understand the test.